### ISSUE TITLE
Add flag for alternative zarf-config.yaml during package create/deploy

### ISSUE THREAD

**Issue by YrrepNoj** [zarf-dev/zarf#2876]
### Is your feature request related to a problem? Please describe.
Right now, it seems like Zarf only reads the `zarf-config.yaml` from the current working directory. During `zarf package create` and `zarf package deploy` I would like to provide a `zarf-config.yaml` that is a different directory than my current working directory.

For my situation I have the following directory structure:
```
packages
├── api
...
...
...
└── whisper
    ├── chart/
    ├── Dockerfile
    ├── README.md
    ├── zarf-config.yaml
    └── zarf.yaml
```

If I am at the root of my repository and perform the command `zarf package create packages/whisper --confirm` then the package that gets created will NOT utilize the values that are specified in the `packages/whisper/zarf-config.yaml`. It is not until I `cd packages/whisper && zarf package create . --confirm` that the values in the config file are utilized.

I would like to be at the root of my repository and perform `zarf package create packages/whisper --config=packages/whisper --confirm` 

### Describe the solution you'd like

- **Given** a user is trying to create/deploy a package
- **When** they provide a flag with a `/path/to/a/zarf-config.yaml`
- **Then** that `zarf-config.yaml` is used.

### Describe alternatives you've considered
1) Right now, I think I am going to have to add `cd some/dir` before my `zarf package create` commands in CI & build scripts. This works but does not feel great.

2) Another alternative is for Zarf to automatically look for config files that are in the same directory as the `zarf.yaml`. The reason why I'm not asking for this in this issue is because that might cause unexpected side affects for users who are have already gotten used to the current functionality.

### Additional context
⚠️ If the  `--config` flag is provided and a `zarf-config.yaml` exists in the current working directory Zarf will need to combine the values in both files (and handle situations where both files set the same variable) and not just ignore one.

**Comment by phillebaba** [comment]
I agree with your conclusion that looking for the config file in the same direction as the build directory could have unintended consequences for people who did not expect this behavior before. Adding a new option might be the best option and would work for other use cases also.

We may have to complete #2682 to make this possible to test.

**Comment by AustinAbro321** [comment]
This is quite similar to #2178 for which there is a PR #2328 though it has been inactive as of late. 

Another option you have is to set the ZARF_CONFIG environment variable to point to your config file. https://docs.zarf.dev/ref/config-files/#_top

> ⚠️ If the --config flag is provided and a zarf-config.yaml exists in the current working directory Zarf will need to combine the values in both files (and handle situations where both files set the same variable) and not just ignore one.

IMO this would be unexpected for users. Maybe a cleaner way to get this behavior would be to make a follow up PR after #2328 to have the flag accept a comma separated list of config files? Even then I'm not sure it's a feature we'd want to add as which file takes priority may get confusing and I'm not sure viper offers a clean way to do this (though they might)

**Comment by brandtkeller** [comment]
Coming back to this issue after the existence of the `package values` alpha feature. We are hopeful that users are migrating package deployment configuration to package values files and using the `zarf-config` file as a CLI runtime configuration file. 

Given that separation there still _may_ be good reasons to want this feature but I agree with the assessment that merging config files adds complexity we may not want to introduce. CLI runtime configuration may need to be a single-file operation and   use of the environment variable establishes optionality to define appropriate coupling of package-in-a-nested-directory with a specific `zarf-config`.

I'm going to close this as not-planned  given #2178 as a duplicate.
